include/oneapi/tbb/info.h

```diff
@@ -57,6 +58,41 @@ struct constraints {
         core_type = id;
         return *this;
     }
+    constraints& set_core_types(const std::vector<core_type_id>& ids) {
+        if (ids.empty()) {
+            core_type = -1;
+            return *this;
+        }
+
+        // Set a marker bit to indicate multiple core type format
+        core_type = (1 << core_type_id_bits);
```

Comment:
What if ids is only size of 1. Should the new format be used or the old. Seems like it would be slightly faster to just set core_type to ids[0] if the size is 1 and keep the old format.